docs/spec2006-tutorial.md

```diff
@@ -0,0 +1,587 @@
+```sh
+deactivate
+```
+### Running the Database Server
+The following script will run the MongoDB database server in a docker container.  
+
+The -p 27017:27017 option maps the port 27017 in the container to port 27017 on the host.  
+The -v /path/in/host:/data/db option mounts the /data/db folder in the docker container to the folder /path/in/host in the host.  
+The path of the host folder should an absoblute path, and the database files created by MongoDB will be in that folder. 
+The --name mongo-1 option specifies the name of the docker container. 
+We can use this name to identify to the container. 
+The -d option will let the container run in the background.  
+mongo is the name of [the offical mongo image](https://hub.docker.com/_/mongo).  
+```sh
+docker run -p 27017:27017 -v /path/in/host:/data/db --name mongo-1 -d mongo
+```
+### Running Celery Server 
+Inisde the path in the host specified above,
+```sh
+celery -E -A gem5art.tasks.celery worker --autoscale=[number of workers],0
+```
+### Creating the Launch Script Running the Experiment  
+Now, we can put together the run script! 
+In launch_spec_experiments.py, we import the required modules and classes at the beginning of the file,
+```python
+import os
+import sys
+from uuid import UUID
+
+from gem5art.artifact import Artifact
+from gem5art.run import gem5Run
+from gem5art.tasks.tasks import run_gem5_instance
+```
+And then, we put the launch function at the end of launch_spec_experiments.py,
+```python
+if __name__ == "__main__":
+    cpu = ['kvm', 'atomic', 'o3']
+    benchmark = ['401.bzip2(base)', '403.gcc(base)', '464.h264ref(base)']
+
+for cpu in cpus:
+    for benchmark in benchmarks:
+		      run = gem5Run.createFSRun(
+			        'gem5/build/X86/gem5.opt',
+			        'configs-npb-tests/run_spec.py',
+			        gem5_binary, gem5_repo, experiments_repo,
+			        'linux-stable/vmlinux-4.19.83',
+			        'disk-image/spec2006/spec2006-image/spec2006',
+			        linux_binary, disk_image,
+			        cpu, benchmark
+		      )
+        run_gem5_instance.apply_async((run,))
+```
+### Getting the Results  
+TODO
+
+## References
+[1]  “Standard Performance Evaluation Corporation,” *SPEC CPU® 2006*. [Online]. Available: https://www.spec.org/cpu2006/. [Accessed: 12-Nov-2019].
+
+[2] J. L. Henning, “SPEC CPU2006 benchmark descriptions,” *ACM SIGARCH Computer Architecture News*, vol. 34, no. 4, pp. 1–17, Jan. 2006.
+
+[3] J. Owens, W. Dally, R. Ho, D. Jayasimha, S. Keckler, and L.-S. Peh, “Research Challenges for On-Chip Interconnection Networks,” IEEE Micro, vol. 27, no. 5, pp. 96–108, 2007.
+
+[4] M. McCool, A. Robison, and J. Reinders, Structured parallel programming: patterns for efficient computation. Waltham, MA: Morgan Kaufmann, 2012.
+
+[5] J. L. Hennessy and D. A. Patterson, “A new golden age for computer architecture,” *Communications of the ACM*, vol. 62, no. 2, pp. 48–60, 2019.  
+ 
+
+## Apprendix I. Working SPEC 2006 Benchmarks
```

Comment:
*Very* good! I think this file is the right place to document what works and what doesn't.

I **LOVE** this document!